### ISSUE TITLE
required .htaccess file appears to not work issue

### ISSUE THREAD

**Issue by Pierre-CHABREDIER** [WillyXJ/facileManager#335]
Hello All, Hello WillyXJ,

I upgraded my facileManager today but when I tried to go to the UI I got the following error :

> The required .htaccess file appears to not work with your Apache configuration which is required by facileManager. AllowOverride None in your configuration may be blocking the use of .htaccess.

But in fact rewrite is activated and allowed (and was previously working with older facileManager version) :

```root@*********:~#
apache2ctl -M
Loaded Modules:`
[...]
rewrite_module (shared)
[...]
```
```
root@*********:~# cat /etc/apache2/sites-enabled/facilemanager.conf
<VirtualHost *:80>
[...]
        DocumentRoot /var/www/facileManager
        <Directory /var/www/facileManager/>
                Options -Indexes +FollowSymLinks +SymLinksIfOwnerMatch -MultiViews
                AllowOverride All
                Order allow,deny
                allow from all
        </Directory>
[...]
</VirtualHost>

<IfModule mod_ssl.c>
        <VirtualHost *:443>
[...]
                DocumentRoot /var/www/facileManager
                <Directory /var/www/facileManager/>
                        Options -Indexes +FollowSymLinks +SymLinksIfOwnerMatch -MultiViews
                        AllowOverride All
                        Order allow,deny
                        allow from all
                </Directory>
[...]
        </VirtualHost>
</IfModule>
```

So the only solution I have found is to comment rewrite tests :

```
root@*********:~# vim /var/www/facileManager/fm-includes/init.php
[...]
        /** Test rewrites */
        /*if (!defined('INSTALL')) {
                if (dns_get_record($_SERVER['SERVER_NAME'], DNS_A + DNS_AAAA)) {
                        $test_output = getPostData($GLOBALS['FM_URL'] . 'admin-accounts.php?verify', array('module_type' => 'CLIENT'));
                        $test_output = isSerialized($test_output) ? unserialize($test_output) : $test_output;
                        if (strpos($test_output, _('Account is not found.')) === false) {
                                $message = sprintf(_('The required .htaccess file appears to not work with your Apache configuration which is required by %1s. '
                                                . 'AllowOverride None in your configuration may be blocking the use of .htaccess.'), $fm_name);
                                if ($single_check) {
                                        bailOut($message);
                                } else {
                                        $requirement_check .= displayProgress(_('Test Rewrites'), false, 'display', $message);
                                        $error = true;
                                }
                        } else {
                                if (!$single_check) $requirement_check .= displayProgress(_('Test Rewrites'), true, 'display');
                        }
                }
        }*/
[...]
```

So please can you help me to understand why rewrite tests are not working ?

Best regards.

PS : Sorry for my bad English.

**Comment by WillyXJ** [comment]
Hello,

This is an odd behavior especially if it worked with a previous version of fM.  When you extracted/copied the updated files, was the .htaccess file copied as a UNIX file?  I'm wondering if there are DOS characters which prevents Apache from using it.  Are there any errors in your Apache logs?

**Comment by Pierre-CHABREDIER** [comment]
Hello,

Yes .htaccess is well copied. I even tried to use the old one from previous version, but I have the same error.

For Apache, there is no errors in any logs.

**Comment by WillyXJ** [comment]
In the block of code you commented out in fm-includes/init.php, could you please dump $test_output?  It should look like this:

```
...
                        $test_output = getPostData($GLOBALS['FM_URL'] . 'admin-accounts.php?verify', array('module_type' => 'CLIENT'));
                        $test_output = isSerialized($test_output) ? unserialize($test_output) : $test_output;
                        var_dump($test_output);
...
```

And send me the result?

**Comment by Pierre-CHABREDIER** [comment]
Sure,

Here is the result :

`string(22) "Account is not found. "`

**Comment by Pierre-CHABREDIER** [comment]
Okay, in fact if I modify following line :

`if (strpos($test_output, _('Account is not found.')) === false) {`

To 

`if (strpos($test_output, 'Account is not found.') === false) {`

Then it works, but maybe it brakes something (I'm a noob in PHP).

**Comment by WillyXJ** [comment]
Thank you for the information.  I have been able to reproduce this using an international language.  The change you made might be the correct one, but I need to review the code some more.

**Comment by WillyXJ** [comment]
This is fixed in fM v2.3.1.

**Comment by SerafinSahary** [comment]
Hi,

for reference, if somebody else face such issue in the future

I got similar case today - in theory everything was perfect in Apache configuration but facileManager was still complaining about AllowOverride value.
In my case the issue was with this: dns_get_record($_SERVER['SERVER_NAME'], DNS_A + DNS_AAAA). There were no valid dns name entry for server name. Simple /etc/hosts entry solved the issue
